integration-tests/common.go

```diff
@@ -2,77 +2,173 @@ package integration_tests
-type StarkNetNetworkConfig struct {
-	ContractsDeployed         bool          `mapstructure:"contracts_deployed" yaml:"contracts_deployed"`
-	L1BridgeAddr              string        `mapstructure:"l1_bridge_addr" yaml:"l1_bridge_addr"`
-	Name                      string        `mapstructure:"name" yaml:"name"`
-	ChainID                   int64         `mapstructure:"chain_id" yaml:"chain_id"`
-	URL                       string        `mapstructure:"url" yaml:"url"`
-	URLs                      []string      `mapstructure:"urls" yaml:"urls"`
-	Type                      string        `mapstructure:"type" yaml:"type"`
-	PrivateKeys               []string      `mapstructure:"private_keys" yaml:"private_keys"`
-	ChainlinkTransactionLimit uint64        `mapstructure:"chainlink_transaction_limit" yaml:"chainlink_transaction_limit"`
-	Timeout                   time.Duration `mapstructure:"transaction_timeout" yaml:"transaction_timeout"`
-	MinimumConfirmations      int           `mapstructure:"minimum_confirmations" yaml:"minimum_confirmations"`
-	BlockGasLimit             uint64        `mapstructure:"block_gas_limit" yaml:"block_gas_limit"`
+const (
+	gethClient      = "Hardhat"
+	devnetClient    = "starknet-dev"
+	chainLinkClient = "chainlink"
+	ChainName       = "starknet"
+	ChainId         = "devnet"
+)
+
+var (
+	err error
+	t   *Test
+)
+
+type Test struct {
+	sc         *StarkNetDevnetClient
+	cc         *ChainlinkClient
+	mockServer *ctfClient.MockserverClient
+	Env        *environment.Environment
 }
 
-// GetStarkNetURLs gets remote L1 URL and a local L2 URL
-func GetStarkNetURLs(e *environment.Environment) ([]*url.URL, error) {
-	var urls []*url.URL
-	l2URLs, err := e.Charts.Connections("starknet").LocalURLsByPort("http", environment.HTTP)
-	if err != nil {
-		return nil, err
-	}
-	urls = append(urls, l2URLs...)
-	l1URLs, err := e.Charts.Connections("geth").RemoteURLsByPort("http-rpc", environment.HTTP)
-	if err != nil {
-		return nil, err
-	}
-	urls = append(urls, l1URLs...)
-	return urls, nil
+type StarkNetDevnetClient struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	cfg    *blockchain.EVMNetwork
+	client *resty.Client
 }
 
-func NewStarkNetClient(cfg *StarkNetNetworkConfig, urls []*url.URL) (*StarkNetClient, error) {
+type ChainlinkClient struct {
+	nKeys          []ctfClient.NodeKeysBundle
+	chainlinkNodes []*client.Chainlink
+	bTypeAttr      *client.BridgeTypeAttributes
+	bootstrapPeers []client.P2PData
+}
+
+func (t *Test) NewStarkNetDevnetClient(cfg *blockchain.EVMNetwork) (*StarkNetDevnetClient, error) {
 	ctx, cancel := context.WithCancel(context.Background())
-	c := &StarkNetClient{
+	t.sc = &StarkNetDevnetClient{
 		ctx:    ctx,
 		cancel: cancel,
 		cfg:    cfg,
-		urls:   urls,
-		client: resty.New().SetBaseURL(urls[0].String()),
+		client: resty.New().SetBaseURL(t.GetStarkNetAddress()),
+	}
+	// Currently not needed since we are testing on L2
+	// if err := c.init(); err != nil {
+	// 	return nil, err
+	// }
```

Comment:
This logic should probably not be part of this constructor, but a specific test should do it if it requires it + the fn name `init` should probably be made less generic.

Reply:
Added `ConfigureL1Messaging()` func